sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/SqlTransform.java

```diff
@@ -175,13 +195,32 @@ private void registerFunctions(BeamSqlEnvBuilder sqlEnvBuilder) {
    * </ul>
    */
   public static SqlTransform query(String queryString) {
-    return builder()
-        .setQueryString(queryString)
-        .setUdafDefinitions(Collections.emptyList())
-        .setUdfDefinitions(Collections.emptyList())
-        .setTableProviderMap(Collections.emptyMap())
-        .setAutoUdfUdafLoad(false)
-        .build();
+    return query(queryString, Collections.emptyMap(), QueryParameterMode.NONE);
+  }
+
+  /**
+   * Same as {@link #query(String)} but with named query parameters. Currently only supported by the
+   * ZetaSQL dialect.
+   *
+   * @param queryParameters type depends on the SQL dialect. For ZetaSQL, a {@code Map<String,
+   *     com.google.zetasql.Value>} keyed by the parameter name.
+   */
+  @Experimental
+  public static SqlTransform queryWithNamedParameters(String queryString, Map queryParameters) {
```

Comment:
Stylistically, `SqlTransform` would do better to have public builder methods. Is there a technical barrier? In other words, an API like `SqlTransform.query(...).withParameters(...)` or even `.withNamedParameters` separate from `withPositionalParameters`.

Reply:
(this is https://beam.apache.org/contribute/ptransform-style-guide/#fluent-builder-methods-for-setting-parameters)

Reply:
> Stylistically, SqlTransform would do better to have public builder methods.

Do you mean just for parameters, or overall?